src/autolabel/tasks/attribute_extraction.py

```diff
@@ -354,14 +354,19 @@ def parse_llm_response(
                             )
                             llm_label.pop(attribute["name"], None)
                     elif attr_type == TaskType.MULTILABEL_CLASSIFICATION:
-                        attr_labels = attr_label.split(self.config.label_separator())
-                        for label in attr_labels:
-                            if label not in attr_options:
-                                logger.warning(
-                                    f"Attribute {label} from the LLM response {llm_label} with value {attr_label} is not in the labels list"
-                                )
-                                llm_label.pop(attribute["name"], None)
-
+                        original_attr_labels = attr_label.split(
+                            self.config.label_separator()
+                        )
+                        filtered_attr_labels = list(
+                            filter(lambda x: x in attr_options, original_attr_labels)
+                        )
+                        llm_label[attribute["name"]] = (
+                            self.config.label_separator().join(filtered_attr_labels)
+                        )
+                        if len(filtered_attr_labels) != len(original_attr_labels):
+                            logger.warning(
```

Comment:
can we also log which labels were removed?

Reply:
logging filtered list now